src/ctl/ctl.c

```diff
@@ -102,6 +122,10 @@ static const struct ctl_node *ctl_find_node(const struct ctl_node *nodes,
      * in the main ctl tree.
      */
     while (node_name != NULL) {
+        *name_offset = (int)(node_name - parse_str);
+        if (n != NULL && n->type == CTL_NODE_SUBTREE) {
```

Comment:
add comment for this condition

Reply:
Done

Reply:
please improve grammar
"if a subtree occurs, the subtree handler should be called"

Reply:
Done